drivers/flash/flash_stm32.c

```diff
@@ -329,6 +329,27 @@ static int stm32_flash_init(struct device *dev)
 	LL_AHB3_GRP1_EnableClock(LL_AHB3_GRP1_PERIPH_HSEM);
 #endif /* CONFIG_SOC_SERIES_STM32WBX */
 
+#if defined(CONFIG_SOC_SERIES_STM32F4X)
```

Comment:
Could have a plain `#ifdef`.

Instead of putting the issue number into the commit title, you can add this above the `Signed-off-by:` instead (and explain what the commit does in the title instead):

    Fixes: #18263 

Then the issue will be closed automatically when this gets merged too.